includes/class-vindi-payment.php

```diff
@@ -419,14 +419,23 @@ protected function build_product_items_for_bill($order_item)
             return false;
         }
 
-        return [
+        $item = [
             'product_id'        => $order_item['vindi_id'],
             'quantity'          => $order_item['qty'],
             'pricing_schema'    => [
                 'price'             => $order_item['price'],
                 'schema_type'       => 'per_unit'
             ]
         ];
+
+        if($order_item['type'] == 'discount') {
+            $item = [
```

Comment:
Isso mesmo @DNA, ele sobrescreve o array se o produto for do tipo 'desconto'. Foi a forma que eu encontrei para resolver o problema sem gerar dois arrays diferentes.